### ISSUE TITLE
Minecolonies farmer / bow guards not working

### ISSUE THREAD

**Issue by TwistedSanta** [Thyriel/Minecraft-AEONs#4]
Many workers wont work due to not recognizing available tools.
No iron pick for level 2 builder and no recognized hoe for farmer discovered so far.

**Comment by Thyriel** [comment]
Basically you should be able use PrimalCores flint tools on level 1 workers and tinker's construct tools with the apropiate mining level on level 2+
For example builders needing an iron+ pickaxe work with any Tinker Construct Pickaxe that has mining level iron+ (tested with iron ore in building area and a manasteel pickaxe given to the lvl2 builder)

Only farmer (have to say i never used that one yet) seems to have a serious problem, as somehow he is neither accepting the primalcore flint hoe nor any Tinker mattock. 
I will do a detailed further testing to see if anymore worker / tool combinations have problems beside farmer and look what i can do about it. (If it's only hoe i might re-enable vanilla hoes until minecolonies fixes that)

**Comment by Thyriel** [comment]
In the meantime you could use this command to bypass the farmer problem (as the hoe is hidden in JEI):
`/give @p iron_hoe`

or replace "iron" by whatever material level you need (stone, diamond, golden)

**Comment by psycic21** [comment]
Hello, I am the host of the server Twisted_santa is on.
The issue here is more that minecolonies colonists do not usually play nice with non-vanilla tools. I have given the workers primalcore tools and tinkers' construct tools (I've had this mod before and already knew it wouldn't accept the tinkers' but no harm in trying again right) and the primalcore worked until hitting level 2 when they requested iron level tools and they would not accept any type of tinkers' tools.

On top of this, for some reason they do not recognize vanilla iron tools as proper tools of iron level.

In a nutshell, they work with primalcore, but not with tinkers' construct tools and for some really odd reason vanilla tools. Sorry to jump in on this just figured I'd give you a better idea of what is going on.

**Comment by Thyriel** [comment]
The strange thing is, Tinker tools work on my server (and they should work with minecolonies since a while. I know for example that "simple life 2" had/has a minecolonies version in 1.10 running where TiC tools did not work, but in 1.12 version they should work) with all kind of workers. Only exceptions i found so far, with a fully upgraded lvl2 colony, is farmer and the guard that needs a bow.

The reason why they don't accept cheated vanilla tools as proper mining level is because there still is a config that sets them all to mining level stone (still there from an old version where i hadn't removed them from loottables). You could delete the file "config/tool_progression/tool_overwrites.cfg" to give them vanilla mining levels again. But, this doesn't touch hoes. If the farmer isn't working with cheated iron vanilla hoes (works on my server with)removing that file wouldn't fix the farmer problem.

I can give you a detailed list of what workers i tested with what tools at what levels later if it helps.

**Comment by Thyriel** [comment]
I've tested now (on server just to be sure it's not a mp problem) with a new colony from lvl 1 to 3 and all buildings (i even let my lvl3 builder build a house inside a mountain to test through tools) and nearly all do work with tinker shovels, hatchets, lumberaxes or pickaxes (even works when stored in warehouse and delivered through deliveryman) and standard configurations from the modpack.

If it still doesn't work for you please try/check the following things:
- minecolonies version is minecolonies-universal-1.12.2-0.8.4280
- forge used is 14.23.0.2512
- disable auto hiring in town hall. Then kill all colonists, wait for new ones to spawn and hire them manually. (Helped me once in an older version when my miner somewhen refused to work anymore for no reason)
- try different mining levels. Sometimes the displayed request text is buggy (old minecolony problem). Had it often happening that they request maximum / minimum stuff but mean it the other way around.

In the next update i will reenable (likely with a custom recipe) vanilla hoes and bow/arrows to fix the farmer / bow guards

**Comment by psycic21** [comment]
I have done the steps above multiple times over. Minecolonies is the correct version according to what the pack should have, same with forge. I have hired, rehired, killed, respawned, rehired, reformed town, burned to the ground and danced on the ashes. I have given them one pickaxe of every mining level available in the game, then gave them different material pickaxes for good measure. I can't explain why, but it has not been working. Thanks for your input, I'm sure I'll figure out what is wrong eventually and fix it in either config or code. Was just hoping you had a simpler answer.

**Comment by Thyriel** [comment]
The thing is, i did not change anything in the config code / scripts that would affect minecolonies and tinker tool handling.
Only things changed is the mining level of a lot tinker materials but that shouldn't affect the usability of tools.
Does it work if you give them vanilla tools after deleting "config/tool_progression/tool_overwrites.cfg" ?

**Comment by Thyriel** [comment]
Maybe it's something in the server settings that makes the difference between your and my server.
allow-flight true (needed for davincis vessels, shouldn't affect minecolonies)
enable-query true and enable-rcon true (don't think these should do any difference but maybe worth a try)
enable-command-block true (worth a try if you have it disabled)
spawn-npcs true
spawn-protection (just make sure spawn protected area doesn't overlap with the colony = colony center more then 100 blocks away (but no more then 8000 blocks)  from server spawn point)

**Comment by Thyriel** [comment]
1.4 released that should fix the farmer / archer (reenabled vanilla hoes/bows/arrows with custom recipes)
https://minecraft.curseforge.com/projects/minecraft-aeons/files/2495748

This patch also updates minecolonies to the newest version (just in case it may fix the TiC tools not working problem for you, usually i prefer beta over alpha version of mods)

And (not in the patchnotes) it removes the tweaked mining level of the hidden vanilla tools (axe, pickaxe, etc.) so in case TiC tools still not work for you, it should at least work if you cheat vanilla tools with a command and give them to the workers.
